WcaOnRails/app/jobs/submit_results_nag_job.rb

```diff
@@ -0,0 +1,14 @@
+class SubmitResultsNagJob < ActiveJob::Base
+  queue_as :default
+
+  def nag_needed(competition)
+    (competition.results_nag_sent_at || competition.end_date) <= 1.week.ago
+  end
+
+  def perform()
+    Competition.where(results_posted_at: nil).all.select { |c| nag_needed(c) }.each do |competition|
+      competition.update_attribute(:results_nag_sent_at, Time.now)
+      CompetitionsMailer.submit_results_nag(competition).deliver_later
```

Comment:
Isn't it simpler to send the email here and now? There is no user waiting.

It would also make the `results_nag_sent_at` value more true.